superset/commands/dashboard/importers/v0.py

```diff
@@ -252,20 +255,30 @@ def alter_native_filters(dashboard: Dashboard) -> None:
 
     alter_native_filters(dashboard_to_import)
 
-    new_slices = (
-        session.query(Slice).filter(Slice.id.in_(old_to_new_slc_id_dict.values())).all()
-    )
-
     if existing_dashboard:
         existing_dashboard.override(dashboard_to_import)
-        existing_dashboard.slices = new_slices
-        session.flush()
-        return existing_dashboard.id
+    else:
+        db.session.add(dashboard_to_import)
+
+    dashboard = existing_dashboard or dashboard_to_import
+    dashboard.slices = (
+        db.session.query(Slice)
+        .filter(Slice.id.in_(old_to_new_slc_id_dict.values()))
+        .all()
+    )
 
-    dashboard_to_import.slices = new_slices
-    session.add(dashboard_to_import)
+    # Migrate any filter-box charts to native dashboard filters.
+    migrate_dashboard(dashboard)
     session.flush()
-    return dashboard_to_import.id  # type: ignore
+
+    # Remove all obsolete filter-box charts.
```

Comment:
I originally updated `import_dashboard` and `import_chart` to return `Dashboard` and `Slice` objects respectively (as opposed to the associated ID), however this became a messy yak shaving experience. For the interest of minimizes the footprint of the change, we simply keep track of the IDs associated with the charts added and then refetch them from the session.